Documentation/building/crossgen.md

```diff
@@ -0,0 +1,64 @@
+- If you install crossgen from NuGet, make sure you use crossgen from exactly the same NuGet package as the rest of you CoreCLR binaries. Do not attempt to mix binaries from multiple NuGet packages.
+
+If you do not follow the above rules, you are likely to encounter errors while running crossgen.
+
+Using crossgen
+--------------
+
+In most cases, the build script automatically runs crossgen to create the native image for mscorlib.
+When this happens, you will find boths `mscorlib.dll` and `mscorlib.ni.dll` in your output directory.
+`mscorlib.dll` is the MSIL assembly creating by the C# compiler, while `mscorlib.ni.dll` is the native image that contains CPU-specific code.
+Once the build is done, you only need `mscorlib.ni.dll` to use CoreCLR.
+As a matter of fact, most CoreCLR NuGet packages contain only `mscorlib.ni.dll`, without `mscorlib.dll`
+
+If for some reason you did not get `mscorlib.ni.dll` with the rest of your CoreCLR, you can easily create it yourself using crossgen.
+First, make sure you have `crossgen.exe` (on Windows) or `crossgen` (other platforms) in the same directory as `mscorlib.dll`.
+Then, run one of the following two commands (first command for Windows, second command for other platforms):
+
+    .\crossgen.exe mscorlib.dll
+    ./crossgen mscorlib.dll
+    
+To create native images for other assemblies, the command line is slightly more complex:
+
+    .\crossgen.exe /Platform_Assemblies_Paths "path1;path2;..." assemblyName.dll
+    ./crossgen /Platform_Assemblies_Paths "path1;path2;..." assemblyName.dll
+    
+The /Platform_Assemblies_Paths is used to specify the location of all the dependencies of the input assembly.
+You should use full paths for this locations. Relative paths do not always work.
+If there are multiple paths, separate them with semicolons (`;`). On non-Windows platforms, be sure to enclose the path list in quotes to protect the semicolons from the shell.
+
+Using native images
+-------------------
+
+Running crossgen on an assembly creates a "native image" file, with the extension of `.ni.dll` or `.ni.exe`.
+You should include the native images in your app, at the same location where you normally install the MSIL assemblies.
+Once you have included native images, you do not need to include the original MSIL assemblies in your apps.
+
+Common errors
+-------------
+
+The following are some of the command errors while creating or using native images:
+- "Could not load file or assembly 'mscorlib.dll' or one of its dependencies. The native image could not be loaded, because it was generated for use by a different version of the runtime. (Exception from HRESULT: 0x80131059)": This error indicates that there is a mismatch between crossgen and mscorlib.ni.dll. Make sure use crossgen and mscorlib.ni.dll from the same build or NuGet package.
```

Comment:
... there is a mismatch between CrossGen and...
Make sure **to** use CrossGen...